cunumeric/deferred.py

```diff
@@ -3224,3 +3223,36 @@ def unpackbits(
         scale = tuple(8 if dim == axis else 1 for dim in range(src.ndim))
         task.add_constraint(p_out <= p_in * scale)  # type: ignore
         task.execute()
+
+    @auto_convert([1])
+    def _wrap(self, src: Any, new_len: int) -> None:
+        if src.base.kind == Future:
+            src = self._convert_future_to_store(src)
+        if self.base.kind == Future:
+            self = src._convert_future_to_store(self)
+
+        # first, we create indirect array with PointN type that
+        # (len,) shape and is used to copy data from original array
+        # to the target 1D wrapped array
+        N = src.ndim
+        pointN_dtype = self.runtime.get_point_type(N)
+        indirect = cast(
```

Comment:
Use `create_store` with `optimize_scalar=False` here, to make sure we don't get a `Future`, which Legion copies don't like.

Reply:
I guess this is not necessary, if this function is only meant to be called with `new_len > 1`.